core/frontend/src/tile/ImdlReader.ts

```diff
@@ -223,15 +223,31 @@ export class ImdlReader extends GltfReader {
   }
 
   private async readNamedTexture(namedTex: any, name: string): Promise<RenderTexture | undefined> {
-    const bufferViewId = JsonUtils.asString(namedTex.bufferView);
-    const bufferViewJson = 0 !== bufferViewId.length ? this._bufferViews[bufferViewId] : undefined;
-    if (undefined === bufferViewJson)
-      return undefined;
+    let texBytes: Uint8Array | undefined;
 
-    const byteOffset = JsonUtils.asInt(bufferViewJson.byteOffset);
-    const byteLength = JsonUtils.asInt(bufferViewJson.byteLength);
-    if (0 === byteLength)
-      return undefined;
+    // If the texture has a valid Id64 name, it is valid to request it directly from the backend.
+    // If the texture does not have a valid Id64 name, it is not valid to request it directly from the backend.
+    // In that case, we instead retrieve the texture directly from the tile.
+    // Furthermore, if external textures are disabled, we retrieve the texture directly from the tile in both cases.
+
+    const useExternalTexture = Id64.isValidId64(name) && IModelApp.tileAdmin.enableExternalTextures;
+    if (useExternalTexture) {
+      texBytes = await this._iModel.getTextureImage(name);

```

Comment:
If `getTextureImage` throws an exception, the entire tile will be discarded. Wrap in try-catch?

Reply:
Good idea.

Reply:
Added this.

Reply:
In the case of an exception, will it continue to display the placeholder texture, or fall back to untextured?

Reply:
It will continue to display placeholder... not sure how to go back to untextured...
should the texture handle get invalidated somehow?

Reply:
@pmconne I do not see a simple way to make it fall back to untextured - perhaps you see a good way.  As it stands, ImdlReader immediately gets back a texture handle to store in its _namedTextures array.  I could perhaps modify the placeholder texture to be a 1x1 fully transparent texture if an exception occurs.

Reply:
I was mostly just curious what would happen. With embedded textures, if we go to embed the texture and discover its ImageSource is invalid, we omit the entire mesh from the tile. Displaying the mesh with a placeholder texture sounds like an improvement to me.